### PR TITLE
Group generated type/op/attr docs by dialect

### PR DESCRIPTION
**Pull request by j2kun**
Fixes https://github.com/google/heir/issues/121

Also added op docs for poly/bgv which didn't have them already

### CODE REVIEW

**Comment by j2kun** [comment]
Example screenshot of the result: 
image

File: include/Dialect/BGV/IR/BUILD [hunk 1]
```diff
@@ -47,6 +47,12 @@ gentbl_cc_library(
             ],
             "BGVDialect.cpp.inc",
         ),
+        (
```
Comment (asraa) [inline include/Dialect/BGV/IR/BUILD:50]: Thanks! This actually reminds me to make a cleanup to create a bazel rule that does all the default creation for types, ops, and dialect. Similar to [`add_mlir_dialect` ](https://mlir.llvm.org/docs/Tutorials/CreatingADialect/#tablegen-targets)

File: include/Dialect/Secret/IR/SecretDialect.td [hunk 2]
```diff
@@ -2,11 +2,13 @@
 #define HEIR_INCLUDE_DIALECT_SECRET_IR_SECRETDIALECT_TD_
 
 include "mlir/IR/DialectBase.td"
+include "mlir/IR/OpBase.td"
 
 def Secret_Dialect : Dialect {
   let name = "secret";
-  let summary = "A dialect for computations that operate on encrypted secrets.";
```
Comment (asraa) [inline include/Dialect/Secret/IR/SecretDialect.td:8]: Should we note somewhere in a style guide not to use `summary`? Was it making docs look bad?

**Review by asraa — APPROVED** [review]

File: include/Dialect/Secret/IR/SecretDialect.td (hunk 2 above)
Comment (j2kun) [reply include/Dialect/Secret/IR/SecretDialect.td:8]: I don't think it's bad, but it just emits the summary as an extra line above the description with no extra line break. So if you expect that then you can make it look right.

**Pull request merged**